SvmTask.py

```diff
@@ -0,0 +1,565 @@
+    print cached_rdd.count()
+    print time.time() - time_0
+    time_0 = time.time()
+    
+    pass
+finally:
+    cached_rdd.unpersist()
+
+
+
+# In[ ]:
+
+
+
+
+# # Notes
+# 
+# 
+#  - You will need to create the output table in Cassandra using a create table statement.
+#  
+#  - To better test the SVM, once code is working, you could change the way the test features and labels are generated so that there is a relationship between the labelled class and the features vector.
+
+# In[ ]:
+
+
+        
+
+
+# In[ ]:
+
+
+
+
+# In[32]:
+
+from pyspark.sql.functions import UserDefinedFunction
+from pyspark.sql.types import StringType
+from pyspark.sql.functions import udf
+from pyspark.sql.types import IntegerType
+from pyspark.sql.functions import col
+
+keys = ['cat', 'dog', 'hamster']
+
+def prepareData():
+    df_feats = read_from_cassandra(sqlContext, my_keyspace, features_table_name)
+    df_labels = read_from_cassandra(sqlContext, my_keyspace, labels_table_name)
+   
+    feat_set_id = df_feats.select(df_feats['feature_set_id']).take(1)[0].feature_set_id
+    print 'Using feature set id : '
+    print feat_set_id
+   
+    # udf_label = udf(getClassLabel, IntegerType())
+    # dx = df_labels.withColumn('class_labels', udf_label(df_labels['labels']))
+    #dx.show()//
+    res_df = df_feats.filter(df_feats['feature_set_id'] == feat_set_id).join(df_labels, df_feats.image_id == df_labels.image_id).select(df_feats['image_id'], df_feats['feature_vector'], df_labels['labels'])
+    #res_df_labeled = res_df.withColumn('classLabel', getClassLabel(res_df.labels))
+    exprs = [col("labels").getItem(k).alias(k) for k in keys]
+    trans_data = res_df.select(res_df.image_id,res_df.feature_vector, *exprs)
+    trans_data = trans_data.na.fill(0)
+   
+    return trans_data
+ 
+
+
+# In[39]:
+
+def createResTable(results_table_name):
+    quick_cql("CREATE TABLE IF NOT EXISTS \"%s\".\"%s\"( query_id uuid, score float, image_id uuid, PRIMARY KEY ((query_id), score) ) WITH caching = { 'keys' : 'ALL', 'rows_per_partition' : '100000' };"% (my_keyspace, results_table_name)
+    )
+    results_schema = StructType([
+        StructField('query_id', StringType(), True),
+        StructField('image_id', StringType(), True),
+        StructField('score', FloatType(), True)
+    ])
+    return results_schema
+
+
+# In[48]:
+
+
+from pyspark.ml import Pipeline
+from pyspark.ml.evaluation import BinaryClassificationEvaluator
+from pyspark.ml.tuning import CrossValidator, ParamGridBuilder
+from pyspark.sql import Row
+from pyspark.sql import functions as F
+from pyspark.mllib.classification import SVMWithSGD, SVMModel
+from pyspark.mllib.regression import LabeledPoint
+from pyspark.sql.functions import col
+
+#Paramters to tune
+my_iterations = [5000];
+my_regParam = [0.01, 0.001]
+my_step = [ 2**j for j in range(-3, -2) ]
+numFolds=3
+
+print 'prepare data'
+trans_data = prepareData()
```

Comment:
trans_data and its derivatives are not cached at any point - how might this help?